src/applications/gqrx/remote_control.cpp

```diff
@@ -902,12 +908,46 @@ QString RemoteControl::cmd_lnb_lo(QStringList cmdlist)
     }
 }
 
+// Dump signal to noise ratio map
+// \dump_map
+QString RemoteControl::cmd_dump_map() const
+{
+    QString answer;
+
+    answer.append(QString("%1\n").arg(snr_map.hfreq));
+    answer.append(QString("%1\n").arg(snr_map.irate));
+    answer.append(QString("%1\n").arg(snr_map.snr.size()));
+    for (size_t i = 0; i < snr_map.snr.size(); i++)
+        answer.append(QString("%1 ").arg((double)snr_map.snr[i], 0, 'f', 1));
+    answer.append(QString("\n"));
+    return answer;
+}
+
+void RemoteControl::populate_map(const std::vector<float> map) {
+    std::vector<float> cache;
+    cache.reserve(map.size());
+
+    for (size_t i = 0 ; i < map.size(); i++) {
+        cache.push_back(20.f * log10f(map[i] / 200.f));
+    }
```

Comment:
This loop reduces Gqrx's performance substantially when the FFT size and/or rate are high, even if the remote control is not used.

I think it will be necessary to defer the calculations until the time that `\dump_map` is requested.

Reply:
Or ditch it and send over raw fft.

Reply:
That would be an improvement, but I suspect the memory copy would still be a drag.

Reply:
how about now?